pkg/backup/backupdest/backup_destination_test.go

```diff
@@ -337,4 +339,368 @@ func TestBackupRestoreResolveDestination(t *testing.T) {
+		t, backuptestutils.MultiNode, backuptestutils.WithParams(params),
+	)
+	defer cleanupFn()
+	db.Exec(t, "SET CLUSTER SETTING backup.index.read.enabled = true")
+	execCfg := tc.Server(0).ApplicationLayer().ExecutorConfig().(sql.ExecutorConfig)
+
+	collections := []string{
+		"nodelocal://1/backup/default?COCKROACH_LOCALITY=default",
+		"nodelocal://2/backup/west?COCKROACH_LOCALITY=region%3Dwest",
+		"nodelocal://3/backup/south?COCKROACH_LOCALITY=region%3Dsouth",
+	}
+
+	var aost1 string
+	db.Exec(t, "BACKUP INTO ($1, $2, $3)", collections[0], collections[1], collections[2])
+	db.QueryRow(t, "SELECT now()").Scan(&aost1)
+	db.Exec(
+		t, "BACKUP INTO LATEST IN ($1, $2, $3) AS OF SYSTEM TIME $4::STRING",
+		collections[0], collections[1], collections[2], aost1,
+	)
+	db.Exec(
+		t, "BACKUP INTO LATEST IN ($1, $2, $3)",
+		collections[0], collections[1], collections[2],
+	)
+	db.Exec(
+		t, "BACKUP INTO LATEST IN ($1, $2, $3)",
+		collections[0], collections[1], collections[2],
+	)
+
+	aost1TS, err := time.Parse(time.RFC3339, aost1)
+	require.NoError(t, err)
+	aost1Hlc := hlc.Timestamp{WallTime: aost1TS.UnixNano()}
+
+	var fullSubdir string
+	db.QueryRow(
+		t, "SHOW BACKUPS IN ($1, $2, $3)",
+		collections[0], collections[1], collections[2],
+	).Scan(&fullSubdir)
+	require.NotEmpty(t, fullSubdir)
+
+	mem := execCfg.RootMemoryMonitor.MakeBoundAccount()
+	defer mem.Close(ctx)
+
+	t.Run("resolve backup manifests with latest AOST", func(t *testing.T) {
+		uris, manifests, locality, memSize, err := indexedResolveBackupManifests(
+			ctx,
+			&mem,
+			collections,
+			execCfg.DistSQLSrv.ExternalStorageFromURI,
+			fullSubdir,
+			hlc.Timestamp{},
+			nil, /* encryption */
+			nil, /* kms */
+			username.RootUserName(),
+			false, /* includeSkipped */
+			true,  /* includeCompacted */
+		)
+		defer mem.Shrink(ctx, memSize)
+		require.NoError(t, err)
+
+		require.Len(t, uris, 4)
+		require.Len(t, manifests, 4)
+		require.Len(t, locality, 4)
+	})
+
+	t.Run("resolve backup manifests with AOST in middle of chain", func(t *testing.T) {
+		uris, manifests, locality, memSize, err := indexedResolveBackupManifests(
+			ctx,
+			&mem,
+			collections,
+			execCfg.DistSQLSrv.ExternalStorageFromURI,
+			fullSubdir,
+			aost1Hlc,
+			nil, /* encryption */
+			nil, /* kms */
+			username.RootUserName(),
+			false, /* includeSkipped */
+			true,  /* includeCompacted */
+		)
+		defer mem.Shrink(ctx, memSize)
+		require.NoError(t, err)
+
+		require.Len(t, uris, 2)
+		require.Len(t, manifests, 2)
+		require.Len(t, locality, 2)
+	})
+
+	t.Run("resolve backup manifests with compacted backups", func(t *testing.T) {
+		// Since backup compactions don't yet support locality aware backups (as of
+		// Aug 2025), we create a custom chain to perform compactions on.
+		// TODO (kev-cao): Update this test to use the previously created chain once
+		// we support locality aware backups in backup compactions.
+		skip.UnderStress(t, "takes too long")
```

Comment:
note: skip doesn't work at the subtest level, it will skip all subsequent subtests i think!